packages/components/src/components/hds/advanced-table/index.ts

```diff
@@ -425,9 +444,71 @@ export default class HdsAdvancedTable extends Component<HdsAdvancedTableSignatur
       this._selectAllCheckbox.indeterminate =
         selectedRowsCount > 0 && selectedRowsCount < selectableRowsCount;
       this._isSelectAllCheckboxSelected = this._selectAllCheckbox.checked;
-      this._selectAllCheckbox.dispatchEvent(
-        new Event('toggle', { bubbles: false })
+    }
+  }
+
+  @action
+  didInsertExpandButton(): void {
+    this.setExpandAllState();
+  }
+
+  @action
+  willDestroyExpandButton(): void {
+    this.setExpandAllState();
+  }
+
+  @action
+  setExpandAllState(): void {
+    if (this._expandAllButton && this._element) {
+      const expandButtons = Array.from(
+        this._element.querySelectorAll(
+          '.hds-advanced-table__tbody .hds-advanced-table__th-button--expand'
+        )
       );
+
+      const parentRowsCount = expandButtons.length;
+      const expandedRowsCount = expandButtons.filter(
+        (button) => button.getAttribute('aria-expanded') === 'true'
+      ).length;
+
+      let expandAllState: boolean | 'mixed';
+
+      console.log('parentRows', expandButtons);
+      console.log(
+        'expandedRows',
+        expandButtons.filter(
+          (button) => button.getAttribute('aria-expanded') === 'true'
+        )
+      );
+
+      if (parentRowsCount === expandedRowsCount) expandAllState = true;
+      if (expandedRowsCount === 0) expandAllState = false;
```

Comment:
nit: should this be an `else if`? doesn't make a different in the grand scheme of things, but it could make sense since the conditionals are related